controllers/topology/reconcile_state.go

```diff
@@ -374,10 +385,63 @@ func (r *ClusterReconciler) reconcileReferencedTemplate(ctx context.Context, in
 	// Set up a cleanup func for removing the old template.
 	// NOTE: This function must be called after updating the object containing the reference to the Template.
 	return func() error {
-		log.Info("Deleting", in.desired.GroupVersionKind().String(), in.desired.GetName())
+		logReconcileAction(log, deleteObjectReconcileOperation, in.current)
 		if err := r.Client.Delete(ctx, in.current); err != nil {
 			return errors.Wrapf(err, "failed to delete %s/%s", in.desired.GroupVersionKind(), in.desired.GetName())
 		}
 		return nil
 	}, nil
 }
+
+type reconcileAction int
+
+const (
+	createObjectReconcileOperation reconcileAction = iota
+	noChangesReconcileOperation
+	patchObjectReconcileOperation
+	rotateTemplateReconcileOperation
+	deleteObjectReconcileOperation
+)
+
+// logReconcileAction takes care of creating consistent log message for reconcile state.
+// NOTE: In order to improve log readability for people using kubectl logs, we are adding
+// a minimal set of information about the object being reconciled to the message as well.
+func logReconcileAction(log logr.Logger, op reconcileAction, obj client.Object, keyAndValues ...interface{}) {
```

Comment:
```suggestion
func logReconcileAction(log logr.Logger, action reconcileAction, obj client.Object, keysAndValues ...interface{}) {
```
nit, if you want (similar as above, to keep it consistent + keys in plural like values)